newIDE/app/src/InstancesEditor/InstancesResizer.js

```diff
@@ -20,98 +61,243 @@ export default class InstancesResizer {
+      roundedTotalDeltaX = this.totalDeltaX;
+      roundedTotalDeltaY = this.totalDeltaY;
+    }
+    if (!isFreeOnX(grabbingLocation)) {
+      roundedTotalDeltaX = 0;
+    }
+    if (!isFreeOnY(grabbingLocation)) {
+      roundedTotalDeltaY = 0;
+    }
+
+    const isLeft =
+      grabbingLocation === 'TopLeft' ||
+      grabbingLocation === 'BottomLeft' ||
+      grabbingLocation === 'Left';
+    const isTop =
+      grabbingLocation === 'TopLeft' ||
+      grabbingLocation === 'TopRight' ||
+      grabbingLocation === 'Top';
+
+    const flippedTotalDeltaX = isLeft
+      ? -roundedTotalDeltaX
+      : roundedTotalDeltaX;
+    const flippedTotalDeltaY = isTop ? -roundedTotalDeltaY : roundedTotalDeltaY;
+
+    let scaleX =
+      (initialSelectionAABB.width() + flippedTotalDeltaX) /
+      initialSelectionAABB.width();
+    let scaleY =
+      (initialSelectionAABB.height() + flippedTotalDeltaY) /
+      initialSelectionAABB.height();
+
+    // Applying a rotation then a scaling can result to
+    // an affine transformation with a shear composite.
+    // So, keeping the aspect ratio ensures a transformation without any shear.
+    let hasRotatedInstance = false;
+    for (let i = 0; i < instances.length && !hasRotatedInstance; i++) {
+      // It only applies to instances that are not strait.
+      hasRotatedInstance = instances[i].getAngle() % 90 !== 0;
+    }
+    if (proportional || hasRotatedInstance) {
+      // Choose the axis where the selection is the biggest.
+      // That way the cursor is always on one edge.
+      if (
+        !flippedTotalDeltaY ||
+        (flippedTotalDeltaX &&
+          flippedTotalDeltaX * initialSelectionAABB.height() >
+            initialSelectionAABB.width() * flippedTotalDeltaY)
+      ) {
+        scaleY = scaleX;
+      } else {
+        scaleX = scaleY;
+      }
+    }
+
+    // No negative scale because instances can't be mirrored
+    // Make the minimal selection size to 1 to avoid absorption
+    scaleX = Math.max(1 / initialSelectionAABB.width(), scaleX);
+    scaleY = Math.max(1 / initialSelectionAABB.height(), scaleY);
+
+    const anchorX =
+      initialSelectionAABB.right -
+      resizeGrabbingRelativePositions[grabbingLocation][0] *
+        initialSelectionAABB.width();
+    const anchorY =
+      initialSelectionAABB.bottom -
+      resizeGrabbingRelativePositions[grabbingLocation][1] *
+        initialSelectionAABB.height();
+
     for (let i = 0; i < instances.length; i++) {
       const selectedInstance = instances[i];
 
-      if (!selectedInstance.hasCustomSize()) {
+      let initialUnrotatedInstanceAABB = this._getOrCreateUnrotatedInstanceAABB(
+        selectedInstance
+      );
+      let initialInstanceOriginPosition = this._getOrCreateInstanceOriginPosition(
+        selectedInstance
+      );
+
+      const angle = ((selectedInstance.getAngle() % 360) + 360) % 360;
+      if (
+        (!proportional && !hasRotatedInstance && angle === 90) ||
+        angle === 270
```

Comment:
I'm not sure I understand these checks and why angle === 270 is a special case here?

Reply:
I think I forgot the parenthesis around the "or" and the formatter made it weird.
I'll fix this.